typing/typetexp.ml

```diff
@@ -533,7 +544,7 @@ and transl_type_aux env ~row_context ~aliased ~policy styp =
             with_local_level_generalize_structure_if_principal begin fun () ->
               let t = newvar () in
               (* Use the whole location, which is used by [Type_mismatch]. *)
-              TyVarEnv.remember_used alias.txt t styp.ptyp_loc;
+              TyVarEnv.remember_used ~check:true alias.txt t styp.ptyp_loc;
```

Comment:
I am wondering if we want to use the alias location `alias.loc` for the warning rather than `styp.ptyp_loc`.

Reply:
This seems indeed a good idea. I replaced the boolean by a optional location is order to do this